### PR TITLE
make password optional for jdbc connection

### PR DESCRIPTION
**Pull request by ajmarcus**
@syodage making password optional for JDBC connection

### CODE REVIEW

File: scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala [hunk 1]
```diff
@@ -84,14 +85,31 @@ package object jdbc {
...
     case JdbcReadOptions(connOpts, query, _, _) => jdbcIoId(connOpts, query)
     case JdbcWriteOptions(connOpts, statement, _) => jdbcIoId(connOpts, statement)
   }
 
+  private def getDataSourceConfig(connOpts: jdbc.JdbcConnectionOptions): DataSourceConfiguration = {
+    val dataSourceConfig = jio.JdbcIO.DataSourceConfiguration
+      .create(connOpts.driverClass.getCanonicalName, connOpts.connectionUrl)
+      .withUsername(connOpts.username)
+    if (connOpts.password.isDefined) {
+      dataSourceConfig.withPassword(connOpts.password.get)
```
Comment (ajmarcus) [inline scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala]: not sure if there is a better way to do this than calling `.get`

Comment (syodage) [reply scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala]: don't see any downside as`.get` is safeguard with `if` check

Comment (andrewsmartin) [reply scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala]: Maybe `connOpts.password.foreach(p => dataSourceConfig.withPassword(p))`

**Comment by syodage** [comment]
Introduce many code style changes, make code style consistent with rest? 
Other than that, LGTM! 👍 
@nevillelyh @ravwojdyla have any comment?

File: scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala [hunk 2]
```diff
@@ -84,14 +85,31 @@ package object jdbc {
     def apply[T](jdbcIoOptions: JdbcIoOptions): JdbcIO[T] = JdbcIO[T](jdbcIoId(jdbcIoOptions))
   }
 
-  private def jdbcIoId(opts: JdbcConnectionOptions, query: String): String =
-    s"${opts.username}:${opts.password}@${opts.connectionUrl}:$query"
+  private def jdbcIoId(opts: JdbcConnectionOptions, query: String): String = {
+    val user = if (opts.password.isDefined) {
+      s"${opts.username}:${opts.password}"
```
Comment (andrewsmartin) [inline scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala]: Should this be `opts.password.get`? You could also do `val user = opts.password.fold(s"${opts.username}")(pass => s"${opts.username}:${opts.password}")` which is more idiomatic Scala.

Comment (ajmarcus) [reply scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala]: yes - i should also add a test for that.

**Comment by ajmarcus** [comment]
tried to undo all the style changes

File: scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala [hunk 3]
```diff
@@ -23,56 +23,57 @@ import com.spotify.scio.Implicits._
...
 import scala.reflect.ClassTag
 
 /**
- * Main package for JDBC APIs. Import all.
- *
- * {{{
- * import com.spotify.scio.jdbc._
- * }}}
- */
+  * Main package for JDBC APIs. Import all.
```
Comment (ravwojdyla) [inline scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala]: Fix comments indents.

File: scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala [hunk 4]
```diff
@@ -154,4 +170,4 @@ package object jdbc {
     }
   }
 
-}
+}
```
Comment (ravwojdyla) [inline scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala]: Add new line.

**Comment by ajmarcus** [comment]
appreciate all the feedback. looking better?

**Pull request merged**